### ISSUE TITLE
add `datetime` as version scheme

### ISSUE THREAD

**Issue by tschmidtb51** [package-url/vers-spec#14]
I suggest to add a version scheme `datetime` that is defined as follows:
To compare 2 versions a and b, do:

Compare the as datetime parsed values (including the timezones).

Use case: Especially in cloud environments, often the version is given by a certain timestamp.

Flagging @pombredanne for attention

**Comment by tschmidtb51** [comment]
This has be implemented

**Comment by tschmidtb51** [comment]
@immqu It looks like testsuite is missing for this one at https://github.com/package-url/vers-spec/tree/main/tests or am I missing something?

**Comment by immqu** [comment]
Yes, the test suite was introduced after the datetime addition; I will implement it. Two questions came up, however:
- Should we restrict the letters ```T``` and ```Z``` to uppercase? RFC3339 allows both, so I don't see the need to restrict the type to use only uppercase. Still, the RFC clearly says that specifications may further limit this, see https://www.rfc-editor.org/info/rfc3339/#section-5.6
- Should the ```:``` be percent-encoded? If a parser only splits at the first ```:``` of a vers string, as specified in how-to-parse, then it is not necessary to use percent-encoding here. I'd therefore suggest to simply keep the rules here as they are.

**Comment by tschmidtb51** [comment]
> Yes, the test suite was introduced after the datetime addition; I will implement it.

Thank you! 🚀 

> Two questions came up, however:
> * Should we restrict the letters `T` and `Z` to uppercase? RFC3339 allows both, so I don't see the need to restrict the type to use only uppercase. Still, the RFC clearly says that specifications may further limit this, see https://www.rfc-editor.org/info/rfc3339/#section-5.6

Yes - IMHO, a standard should have one way of doing something and not 4. Multiple variants shift the burden from the one that creates it (usually a single entity) to the consumers (usually multiple entities / the majority) which is not sustainable nor fair.

> * Should the `:` be percent-encoded? If a parser only splits at the first `:` of a vers string, as specified in how-to-parse, then it is not necessary to use percent-encoding here. I'd therefore suggest to simply keep the rules here as they are.

If possible, I would say that they should not have to be percent-encoded as it makes it easier for humans to read them. However, one need to check that against the URI promise (see https://github.com/package-url/vers-spec/blob/main/docs/specification/standard/specification.md#a-vers-is-a-uri-scheme).
